src/DurableTask.AzureStorage/MessageManager.cs

```diff
@@ -341,13 +353,20 @@ static class TypeNameSerializationHelper
         static readonly Assembly DurableTaskCore = typeof(DurableTask.Core.TaskMessage).Assembly;
         static readonly Assembly DurableTaskAzureStorage = typeof(AzureStorageOrchestrationService).Assembly;
 
-        public static void BindToName(Type serializedType, out string assemblyName, out string typeName)
+        public static void BindToName(ICustomTypeBinder customBinder, Type serializedType, out string assemblyName, out string typeName)
         {
-            assemblyName = null;
-            typeName = serializedType.FullName;
+            if (customBinder == null)
```

Comment:
Looks like this condition is flipped.
```suggestion
            if (customBinder != null)
```